trailblazer/server/api.py

```diff
@@ -77,12 +80,16 @@ def analysis(analysis_id):
         return abort(404)
 
     if request.method == "PUT":
-        status: str | None = request.json.get("status")
-        comment: str | None = request.json.get("comment")
-        is_visible: bool | None = request.json.get("is_visible")
-        store.update_analysis(
-            analysis_id=analysis_id, comment=comment, status=status, is_visible=is_visible
-        )
+        try:
+            analysis_update = AnalysisUpdate.model_validate(request.json)
+            store.update_analysis(
+                analysis_id=analysis_id,
+                comment=analysis_update.comment,
+                status=analysis_update.status,
+                is_visible=analysis_update.is_visible,
```

Comment:
Might be cleaner to pass the entire schema here instead to the `update_analysis` method. Let me know what you think.

Reply:
I think this is nice since it validates the request 👍